analysis_driver/dataset.py

```diff
@@ -344,6 +346,27 @@ def _find_pooling_step_for_artifact(art, expected_pooling_step_name=None, max_it
             )
         return art.input_artifact_list()
 
+    @property
+    def rapid_samples_by_lane(self):
+        if self._rapid_samples_by_lane is None:
+            self._rapid_samples_by_lane = {}
+
+            # TODO: move away from using the Lims API
+            flowcell = set(self.lims_run.parent_processes()).pop().output_containers()[0]
+            for lane, artifact in flowcell.placements.items():
+                if len(artifact.reagent_labels) > 1:
+                    continue  # we don't want to run rapid processing on pools
+
+                assert len(artifact.samples) == 1
+                sample = artifact.samples[0]
+                if sample.udf.get('Rapid Analysis'):
```

Comment:
What data type will this UDF be? If it's a string, this will evaluate to `True` if it's something like `'no'`